### DISCUSSION TITLE
Confused about versions

### DISCUSSION THREAD

**Discussion by fabnavigator** [zwave-js/node-red-contrib-zwave-js#281] [Q&A]
"zwave-js / node-zwave-js" is currently at version 11.9.2.

"npm view zwave-js/node-zwave-js version" returns 11.9.2.

"zwave-js / node-red-contrib-zwave-js" doesn't have the 11.9.2 code from "zwave-js / node-zwave-js".

I would like to use the code from 11.9.2. What do I need to do?

**Comment by marcus-j-davies** [comment]
Hi @fabnavigator 

This is because the installed version of the module you have, is already happy about the driver version currently installed -> `^11.6.0`

The base version is currently 11.6.0:

```
"dependencies": {
		"limiter": "^2.1.0",
		"lodash": "^4.17.21",
		"winston": "^3.10.0",
		"winston-transport": "^4.5.0",
		"zwave-js": "^11.6.0"
	},

```
^ : Allow 11.x.x (will install latest if fresh install)
~ : Allow 11.6.x (will install latest if fresh install)


if you install it fresh - you will get what ever driver version is the latest (and compatible)

by default NPM will install what ever dependencies are the latest (and versions that I state should be allowed)
but updating a parent module - doesn't mean it will also update it's dependencies (unless I state a new base version of course)

to force update the driver...
```
cd ~/.node-red
npm update zwave-js
systemctl restart node-red
```

this will pull down whatever is the latest (and compatible) -> `11.x.x `

bear in mind - if you want to cash in on something new that is in 11.9 - it wont, as my code base, wouldn't have been developed for it yet (this excludes config file updates)
but for bug fixes it should be  - as that is driver side code that doesn't offer any advantages to the front end

**Reply by fabnavigator** [reply]
@marcus-j-davies, Thanks for explaining that. I made a driver change that's in 11.9.2 that I want to use. Now I know how to do that.

**Reply by marcus-j-davies** [reply]
I don't think it made 11.9.2
that was released yesterday - your PR was merged 5 hours ago.

so the next version of the driver will get  it

**Reply by fabnavigator** [reply]
Well, that didn't get me what I wanted. I still have the old driver code.

**Reply by fabnavigator** [reply]
Got it. Thanks.

**Reply by marcus-j-davies** [reply]
double click the node - check versions

**Reply by marcus-j-davies** [reply]
Cross talk 😆

**Reply by marcus-j-davies** [reply]
What ever is in the Main Branch (or merged into main at least) is what is released next.

**Reply by marcus-j-davies** [reply]
Alos remmeber to remove you custom config files. -so you can be sure

**Reply by fabnavigator** [reply]
I see my change in commits to master since this release. I'm all good.
